crates/dekaf/src/utils.rs

```diff
@@ -33,7 +36,26 @@ impl CustomizableExtractor {
                     None => 0,
                 };
 
-                Err(Cow::Owned(serde_json::json!({"is_deleted": deletion})))
+                Err(Cow::Owned(serde_json::json!(deletion)))
+            }
+            CustomizableExtractor::RootExtractorWithIsDeleted => {
+                let deletion = match META_OP_PTR.query(doc) {
+                    Some(n) => match n.as_node() {
+                        doc::Node::String(s) if s == "d" => 1,
+                        _ => 0,
+                    },
+                    None => 0,
+                };
+
+                let mut full_doc = doc.to_debug_json_value();
```

Comment:
The alternative is to figure out how to plumb through an allocator here and turn `doc` into a `HeapNode`... but unless I'm missing something, this seems fine as a stop-gap -- this codepath goes away once everyone migrates over to materializations and we turn off support for that connection method.